ui/app/js/controllers/required_signoffs_controller.js

```diff
@@ -43,6 +43,33 @@ function($scope, $modal, $q, ProductRequiredSignoffs, PermissionsRequiredSignoff
     }
   }, true);
 
+  $scope.undoRemoveRole = function(role, rs, csrf_token) {
+    var roleName = role.$key;
+    var product = Object.keys(rs)[0];
+    var chans = rs[product].channels;
+    var channel = Object.keys(chans)[0];
+
+    $scope.required_signoffs[product].channels[channel][roleName].sc.signoffs_required = 1;
+    $scope.required_signoffs[product].channels[channel][roleName].sc.change_type = "update";
+    var update_sc_id = $scope.required_signoffs[product].channels[channel][roleName].sc.sc_id;
+    var data = {
+      "product": $scope.product, 
+      "role": roleName,
+      "data_version": $scope.required_signoffs[product].channels[channel][roleName].data_version};
+    data["when"] = new Date().getTime() + 5000;
+    data["sc_data_version"] = $scope.required_signoffs[product].channels[channel][roleName].sc.sc_data_version;
+    data["signoffs_required"] = $scope.required_signoffs[product].channels[channel][roleName].sc.signoffs_required;
```

Comment:
I think you've done a bit more work here than is necessary. The only things that you need to include to delete a scheduled change are sc_id, data_version (which comes from sc_data_version), and csrf_token - so you should be able to remove a bunch of these lines of code.